### ISSUE TITLE
[Feature Request] Support for other SR formats than TID 1500

### ISSUE THREAD

**Issue by WojciechTomalaFT** [OHIF/Viewers#4404]
### What feature or change would you like to see made?

We are facing the problem that OHIF Viewer doesn't support other SR formats than TID 1500. 
![image](https://github.com/user-attachments/assets/3d8de736-35d3-4cd2-9d70-fc169deb5c88)

### Why should we prioritize this feature?

Many studies include the SR modality, which is essential for radiologists and plays a crucial role in their work. The current compatibility is insufficient to meet their needs.

**Comment by MaksymKokora** [comment]
Yes, this topic is also highly requested by us, as it aligns closely with our current priorities and the interests of our stakeholders. We have seen a significant demand for further exploration and development in this area.

**Comment by fedorov** [comment]
Do you know which SR templates you would like to see supported?

It would be more helpful if you could describe what templates, and what kind of support specifically you would like to see.

DICOM defines [hundreds and hundreds of templates](https://dicom.nema.org/medical/dicom/current/output/chtml/part16/PS3.16.html). It is relatively simple to transform those into HTML and show the result (that's what we had in OHIF v2, see example below). Is this something you would like to see? I would say it's a regression that rendering of SR HTML is not available in OHIF v3.

https://viewer.imaging.datacommons.cancer.gov/viewer/1.3.6.1.4.1.14519.5.2.1.6279.6001.191425307197546732281885591780
<img width="1387" alt="image" src="https://github.com/user-attachments/assets/ab7669f2-1e6c-4c02-86bb-bb6545c05018">

**Comment by WojciechTomalaFT** [comment]
Hello @fedorov , 
Mainly I need support for this types of SR templates:
- TID 10011. CT Radiation Dose
- TID 10001 Projection X-Ray Radiation Dose
- TID 5101. Vascular Patient Characteristics
- TID 5100. Vascular Ultrasound Report
- TID 4000. Mammography CAD Document Root

Is it possible to create universal logic which support all the templates?

**Comment by fedorov** [comment]
> Is it possible to create universal logic which support all the templates?

I think this depend on what you want that universal logic to do. If you just want to see a human-readable version of the SR document, as in the screenshot I shared earlier, that is definitely possible.

But at least some of the templates you mentioned may include image-derived measurements (ie, TID 5100 includes [TID 300](https://dicom.nema.org/medical/dicom/current/output/chtml/part16/chapter_A.html#sect_TID_300)), which I would imagine one would want to be displayed on the image. And TID 300 is not the only template that A challenge here is that subordinate templates in SR do not include template identifier. As such, one needs to parse the (sometimes, deeply) nested hierarchy of the SR instance and look for patterns that might hint at the specific subordinate template. Add to this the  broad scope of some of the templates, high variability of implementations, non-compliance, and the task quickly becomes rather difficult.

**Comment by sedghi** [comment]
We have added support ofr text based SR, see one example here [https://viewer.ohif.org/viewer?StudyInstanceUIDs=1.2.276.0.7230010.3.1.2.2155604110.4180.1021041295.21](https://viewer.ohif.org/viewer?StudyInstanceUIDs=1.2.276.0.7230010.3.1.2.2155604110.4180.1021041295.21)
